docs/auth.md

```diff
@@ -0,0 +1,66 @@
+
+Producer SDK defines a few constructor APIs for creating IoT based credentials provider: https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/include/com/amazonaws/kinesis/video/common/Include.h#L525
+
+* createCurlIotCredentialProvider - IoT credential provider that makes backend calls using libCurl client
+* createLwsIotCredentialProvider - IoT credential provider that makes backend calls using libWebSockets client
+
+Same APIs but the ones below are taking in a function to retrieve the current time which is neccessary for higher-layers of the SDK or when applications redefine the system clock functionality.
+
+* createCurlIotCredentialProviderWithTime
+* createLwsIotCredentialProviderWithTime
+
+IoT credential provider makes a synchronous RESTful call to IoT endpoint in order to retrieve or refresh the credentials when needed. The first call will be made during the construction of the provider object which makes the construction of the object "non-prompt". The IoT credential provider implements a simple caching policy described below and returns the cached credentials whenever possible instead of actually making backend calls.
+
+IoT credential provider object is freed calling a single freeIotCredentialProvider API.
+
+By default, C, C++ Producers will compile against the Curl version as the streaming is based on libCurl and the WebRtc will compile against the LWS version as the Signaling in WebRtc is already based on libWebSockets.
+
+#### File based credential provider
+
+Some applications have a separate authentication module running in a different process. The auth module will periodically retrieve the new credentials (potentially integrating with IAM directly) but it's hard to share these credentials via an inter-process communication in a cross-platform fashion. For this, the SDK implements a file-based credential provider. The credentials are shared via a file which can be on a persistent media or a memory mapped file. The auth module will produce the frash credentials into the file and the SDK file based credential provider will read from the file and use the credentials.
+
+Just like the IoT provider case, the Producer SDK defines a few constructor APIs for creating file based credentials provider: https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/include/com/amazonaws/kinesis/video/common/Include.h#L602
+
+* createFileCredentialProvider - File based credential provider that reads the credentials from a specified file which gets periodically updated by an authentication module.
+* createFileCredentialProviderWithTime - Same API which also takes a time functionality.
+
+Similar to IoT credential provider, the file based credentials provider implements a caching policy in order to not read the credential file with each call.
+
+File based credential provider object is freed calling a single freeFileCredentialProvider API.
```

Comment:
Can we add this specific line in other credential providers as well pointing to the appropriate free call?

Reply:
I am not sure what you mean by this. I thought of adding a link but decided to add a link to the public header file where every API is located and just put the names of the APIs here. I mean, I could add the links but I am not sure if it makes it more redable

Reply:
Oh. I mean we can callout the equivalent `free()` calls for other credential providers.